Automattic/woocommerce-payments
# Implement loading state to WooPay express checkout button

Fixes #7295 

#### Changes proposed in this Pull Request

- Add loading and disabled state to WooPay express checkout button.

#### Demo

https://github.com/Automattic/woocommerce-payments/assets/5509901/c82c0eb5-fb75-4071-bca2-454ca925aafe

#### Theme preview

**Dark**

https://github.com/Automattic/woocommerce-payments/assets/5509901/dea43447-2617-4c68-a9a3-f1b0741a4447

**Light**

https://github.com/Automattic/woocommerce-payments/assets/5509901/749bf5bd-1077-4e5a-9198-180858c12eb0

**Outline**

https://github.com/Automattic/woocommerce-payments/assets/5509901/93a72833-4ea5-45fc-b5e6-17b16c4cd3cc

#### Testing instructions

1. Ensure you're using the feature flag `_wcpay_feature_woopay_first_party_auth`. There should be no loading indicator without it.
2. Navigate to the product, cart or checkout pages and click the WooPay button. Ensure there's a loading spinner like in the demo video, and that the button is disabled after click.
3. Experiment with tweaking the button appearance settings (CTA, size, theme) and ensure everything looks as expected.

-------------------


**Post merge**
